pkg/gateway/model_build_lattice_service.go

```diff
@@ -76,13 +76,6 @@ func (t *latticeServiceModelBuildTask) buildModel(ctx context.Context) error {
 		return err
 	}
 
-	if !t.httpRoute.DeletionTimestamp.IsZero() {
-		// in case of deleting HTTPRoute, we will let reconcile logic to delete
-		// stated target group(s) at next reconcile interval
-		glog.V(2).Infof("latticeServiceModuleBuildTask: for HTTPRouteDelete, reconcile tagetgroups/targets at reconcile interval")
-		return nil
-	}
-
```

Comment:
Removed this if condition, need to go ahead to build the TargetGroup model to make target_group_synthesizer could receive this  target group deletion request

Reply:
Will do

Reply:
I found currently we don't have the unit test for  `(t *latticeServiceModelBuildTask) buildModel()` method, that is a large method, I plan to add this unit test in a seperate PR

Reply:
can u create a issue for this? thanks

Reply:
Ok